src/pycsou/opt/solver/lsmr.py

```diff
@@ -0,0 +1,332 @@
+# #############################################################################
+# lsmr.py
+# ========
+# Author : Kaan Okumus [okukaan@gmail.com]
+# #############################################################################
+
+import typing as typ
+
+import numpy as np
+
+import pycsou.abc.operator as pyco
+import pycsou.abc.solver as pycs
+import pycsou.opt.stop as pycos
+import pycsou.runtime as pycrt
+import pycsou.util as pycu
+import pycsou.util.ptype as pyct
+
+
+class LSMR(pycs.Solver):
+    r"""
+    LSMR Method.
+
+    The LSMR method solves the system of linear equations :math:`\mathbf{A}\mathbf{x}=\mathbf{b}` iteratively. If the
+    system is inconsistent, it solves the least-squares problem :math:`\min ||\mathbf{b} - \mathbf{A}\mathbf{x}||_2`.
+    :math:`\mathbf{A}` is a rectangular matrix of dimension m-by-n, where all cases are allowed: m=n, m>n or m<n.
+    :math:`\mathbf{b}` is a vector of length m. The matrix :math:`\mathbf{A}` may be dense or sparse.
+
+    ``LSMR()`` **Parameterization**
+
+    A: :py:class:`pycsou.abc.operator.PosDefOp`
+        Any positive definite linear operator is accepted.
+    damp: float
+        Damping coefficient. Default is 0.
+    atol, btol: float
+        Stopping tolerances. LSMR continues iterations until a certain backward error estimate is smaller than some
+        quantity depending on atol and btol. Default is 1e-6 for both.
+    conlim: float
+        LSMR terminates if an estimate :math:`\text{cond}(\mathbf{A})` exceeds conlim. Default is 1e8.
+    iter_lim: int, optional
+        LSMR terminates if the number of iterations reaches iter_lim. Default is None.
+
+    ``LSMR.fit()`` **Parameterization**
+
+    b: NDArray
+         (..., N) 'b' terms in the LSMR cost function. All problems are solved in parallel.
+    x0: NDArray
+        (..., N) initial point(s). Defaults to 0 if unspecified.
+    restart_rate: int
```

Comment:
I think the `restart_rate` parameter is described in the documentation but is not present in the implementation.

Reply:
Fixed that.